### PR TITLE
Hide the "Message" button in the sidebar if the CreateRooms components should not be shown

### PR DESCRIPTION
**Pull request by dhenneke**
When using `ComponentVisibilityCustomisations` to hide certain features from the UI, the "Message" link in the people section of a room is still visible and leads to an error, if the room creation is [disabled in the homeserver](https://github.com/matrix-org/synapse-user-restrictions).

Before:
<table>
<td>
image
</td>
<td>
image
</td>
</tr>
</table>

After:
image

## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Hide the "Message" button in the sidebar if the CreateRooms components should not be shown ([\#9271](https://github.com/matrix-org/matrix-react-sdk/pull/9271)). Contributed by @dhenneke.

### CODE REVIEW

**Review by SimonBrandner — CHANGES_REQUESTED** [review]
LGTM codewise

Would it be possible to write tests for this change?

**Comment by dhenneke** [comment]
I could try to write a unit test tomorrow.

**Comment by SimonBrandner** [comment]
> I could try to write a unit test tomorrow.

Absolutely awesome! Take the time you need and in case you need anything, be sure to visit https://matrix.to/#/#element-dev:matrix.org

File: test/components/views/right_panel/UserInfo-test.tsx [hunk 1]
```diff
@@ -183,5 +192,21 @@ describe('<UserInfo />', () => {
...
         });
+
+        it('renders the message button', () => {
+            const component = getComponent();
+
+            expect(component.find("MessageButton")).toHaveLength(1);
+        });
+
+        it('hides the message button if the visibility customisation hides all create room features', () => {
+            (shouldShowComponent as jest.Mock).mockImplementation((component) => {
```
Comment (SimonBrandner) [inline test/components/views/right_panel/UserInfo-test.tsx:203]: You should be able to use `mocked()` here

**Review by SimonBrandner — COMMENTED** [review]
Could you please fix the strict TS warnings?

LGTM otherwise

**Comment by dhenneke** [comment]
I added additional test for the component. I never worked with `enzyme` before and was surprised that it was used in this file while other components seem to use `@testing-library/react` which I'm more familiar with. But I hope it's correct in this way.

**Comment by t3chguy** [comment]
@dhenneke enzyme is considered deprecated here, please use RTL instead. Enzyme is blocking our React 18 upgrade

**Review by dbkr — APPROVED** [review]

**Comment by dbkr** [comment]
Hey - apologies for this dropping off our radar. This looks fine now though. I'm assuming we don't have write permission to the fork: could you pull in the changes from develop so all the tests run?

**Comment by dhenneke** [comment]
Neither do I... Maybe @maheichyk could help out?

**Comment by maheichyk** [comment]
> Hey - apologies for this dropping off our radar. This looks fine now though. I'm assuming we don't have write permission to the fork: could you pull in the changes from develop so all the tests run?

@dbkr branch is updated. Could you please have a look and merge is fine?

**Pull request merged**